service/tablet_allocator.hh

```diff
@@ -141,12 +154,14 @@ private:
     table_resize_plan _resize_plan;
     tablet_repair_plan _repair_plan;
     bool _has_nodes_to_drain = false;
+    std::vector<drain_failure> _drain_failures;
 public:
     /// Returns true iff there are decommissioning nodes which own some tablet replicas.
     bool has_nodes_to_drain() const { return _has_nodes_to_drain; }
 
     const migrations_vector& migrations() const { return _migrations; }
-    bool empty() const { return _migrations.empty() && !_resize_plan.size() && !_repair_plan.size();}
+    const std::vector<drain_failure>& drain_failures() const { return _drain_failures; }
+    bool empty() const { return _migrations.empty() && !_resize_plan.size() && !_repair_plan.size() && _drain_failures.empty(); }
```

Comment:
How about `size()`?

Reply:
Right. I think we should drop size(), as it's meaningless. It used to count migrations, but now it's not, and it's used for emptiness, but for that empty() is enough.